esi_leap/common/exception.py

```diff
@@ -72,6 +72,13 @@ class OfferNoTimeAvailabilities(ESILeapException):
                 "time range %(start_time)s, %(end_time)s.")
 
 
+class OfferTimeConflictResourceNode(ESILeapException):
```

Comment:
This should probably be renamed to just 'OfferResourceTimeConflict', since the data model never actually talks about node. I could see putting `%(resource_type)` in the error message: `("Offer on %(resource_type)s %(resource_uuid)s cannot be...")` This lets us potentially support other resource types in the future.

Reply:
I changed the text. I cannot add the resource_type unless I want to pass it in as an extra parameter to the db function solely for the purpose of raising errors. It seems silly to me but if you think it is worth it then I can do so.

Reply:
I think there's a misunderstanding here; a resource_uuid does not unique define a resource; it's a resource_type + resource_uuid. Nothing in the offer/contract code should directly reference a node, only a 'resource' which the resource factory class later instantiates. That probably means that `offer_verify_node_availability` needs to take in `resource_type` and filter based on resource_uuid and resource_type

Reply:
ah I did misunderstand it. Its a quick fix.